### PR TITLE
Store Analytics Events as JSON Blobs in Room

### PR DESCRIPTION
**Pull request by tdchow**
### Summary of changes

- This PR refactors analytics event storage in room to store a single piece of JSON
- This allows us to add new Analytics fields without having to perform Room schema migrations
- Original v4 PR: https://github.com/braintree/braintree_android/pull/1051

### Checklist

 - [ ] Added a changelog entry
 - [ ] Relevant test coverage

### CODE REVIEW

File: BraintreeCore/src/test/java/com/braintreepayments/api/core/AnalyticsClientUnitTest.kt [hunk 1]
```diff
@@ -103,58 +123,63 @@ class AnalyticsClientUnitTest {
...
-    @Test
-    fun writeAnalytics_whenEventNameAndTimestampArePresent_returnsSuccess() {
-        val inputData = Data.Builder()
-            .putString(AnalyticsClient.WORK_INPUT_KEY_EVENT_NAME, eventName)
-            .putLong(AnalyticsClient.WORK_INPUT_KEY_TIMESTAMP, timestamp)
-            .build()
-        val sut = AnalyticsClient(httpClient, analyticsDatabase, workManager, deviceInspector)
-        val result = sut.writeAnalytics(inputData)
-        assertTrue(result is ListenableWorker.Result.Success)
+        // TODO: Fix this test
```
Comment (sshropshire) [inline BraintreeCore/src/test/java/com/braintreepayments/api/core/AnalyticsClientUnitTest.kt:127]: q: Is this a leftover note from the migration? We can remove if the test is passing. It looks identical to the test we have in `main`.

**Review by sshropshire — APPROVED** [review]
🔥

File: BraintreeCore/src/test/java/com/braintreepayments/api/core/AnalyticsClientUnitTest.kt (hunk 1 above)
Comment (tdchow) [reply BraintreeCore/src/test/java/com/braintreepayments/api/core/AnalyticsClientUnitTest.kt]: whoops! i forgot to remove this. great catch!

**Review by sarahkoop — APPROVED** [review]

**Comment by sarahkoop** [comment]
Did we spot check events are coming into FPTI on v5 with this change?

**Comment by tdchow** [comment]
> Did we spot check events are coming into FPTI on v5 with this change?

I haven't checked FPTI events on v5 yet, but I'll add a task to verify events as a part of the pre-release testing.

**Pull request merged**